Adding info on migration from shinytest to shinytest2

jen-machin opened the pull request:
Now that Rich has updated the dashboard page, I added in the info on migrating from shinytest to shinytest2 😄 I had previously done this on another branch but the structure of the pages changed so much that it was easier to just create a new branch and start again to avoid chaos!!

cjrace commented on writing-visualising/dashboards_rshiny.qmd:
```diff
@@ -490,6 +496,43 @@ The expected outputs are saved in a [_snaps/shinytest2](https://github.com/dfe-a
+
+There are a couple of potential issues that you may run into when completing migration or running your tests:
+
+**Chromote**
```
Should this be a heading? (I'd assume h4?)

cjrace commented on writing-visualising/dashboards_rshiny.qmd:
```diff
@@ -490,6 +496,43 @@ The expected outputs are saved in a [_snaps/shinytest2](https://github.com/dfe-a
+
+Shinytest2 makes use of the Chromote package, which needs to connect to your local instance of Google Chrome. Please make sure you have Google Chrome installed before you attempt to use shinytest2, otherwise it will not work. Sometimes your copy of Chrome might be installed in a different place to where Chromote expects to find it, which means that your tests will not be able to run. There is inf ...
+
+**Formatting of line endings**
```
Should this be a heading? (I'd assume h4?)

cjrace commented on writing-visualising/dashboards_rshiny.qmd:
````diff
@@ -490,6 +496,43 @@ The expected outputs are saved in a [_snaps/shinytest2](https://github.com/dfe-a
+
+The migration function will only work if specific line endings (also known as carriage returns) are used in your code. It will not be immediately obvious to you what kind of line endings your code uses as default. If you receive an error that looks like this, or any error that seems to be pointing at the blank space at the end of a line of code, then you may be having this issue:
+
+``` 
````
I don't think this code block is formatting right, should it have highlighting and a little copy symbol in the top right corner? [image from local build on this branch]
![image](https://github.com/dfe-analytical-services/analysts-guide/assets/52536248/71f81c5b-dd9c-4b14-abf3-b840a546f719)

cjrace commented on writing-visualising/dashboards_rshiny.qmd:
```diff
@@ -490,6 +496,43 @@ The expected outputs are saved in a [_snaps/shinytest2](https://github.com/dfe-a
+
+**Chromote**
+
+Shinytest2 makes use of the Chromote package, which needs to connect to your local instance of Google Chrome. Please make sure you have Google Chrome installed before you attempt to use shinytest2, otherwise it will not work. Sometimes your copy of Chrome might be installed in a different place to where Chromote expects to find it, which means that your tests will not be able to run. There is inf ...
```
Scope creep a little - but in that section you refer to here we don't put anything on how to find where your chrome is installed. I wonder if someone has this issue, then it might be worth adding a quick paragraph on how to find that out? (I'm assuming there's a quick way like hit windows and right click chrome to find it or a command you could run or something to show you what path it should be?)

jen-machin replied:
Forgot the command doesn't always work properly so I added a different method of finding out!

jen-machin commented:
Also went totally off tangent and fixed a load of problems with formatting in code blocks and updated the rounding section to include more info (I should have done this in a separate branch, oopsy)

cjrace commented on learning-development/r.qmd:
````diff
@@ -92,15 +92,15 @@ Sometimes renv::restore() can fail, and when in specific renv-controlled project
 2. Specifying the renv library as the install location. It's a bit of a fudge, though these lines are helpful to get the packages from the renv lockfile installed and you running the project when needed:
-
-```
+ 
````
Looks like stray whitespace?

jen-machin replied:
I don't even remember editing this line 🤣

cjrace commented on renv.lock:
Looks like the versions of some packages have gone backwards - was this intentional?

jen-machin replied:
It was not, renv hates me

rmbielby commented on learning-development/r.qmd:
```diff
@@ -430,9 +430,20 @@ This can be done most easily by navigating in RStudio through `Tools > Global op
+
+These slight differences in the floating point numbers behind the scenes can result in unusual rounding errors using round(). If you're interested, [there's more information available in these forum posts](https://stat.ethz.ch/pipermail/r-help/2007-August/138014.html). 
+ 
+To round numbers upwards in a way that is consistent with SQL and other programming languages, you can create a custom function like the one below:
```
I think we should switch this up and recommend using the already created function from dfeR, rather than getting everyone to reproduce the custom function.

So switch this line to "To round numbers upwards..., we recommend using the roundFiveUp() function provided by the dfeR package." Could also potentially put roundFiveUp() as a link to the code in dfeR instead of recreating it here.

cjrace reviewed (approved):
Have made one tiny tiny change to fix the whitespace while I was running it locally already, happy to approve .

Double check that last commit of mine and then I think it's good to merge 😄
